src/getValidators.unit.test.ts

```diff
@@ -11,9 +11,137 @@ const client = createPublicClient({
+
+it('getValidators return all validators (Xai)', async () => {
+  const { isComplete, validators } = await getValidators(client, {
     rollupAddress: rollupAdminLogicAddress,
   });
   expect(validators).toEqual(['0x25EA41f0bDa921a0eBf48291961B1F10b59BC6b8']);
+  expect(isComplete).toBeTruthy();
+});
+
+it('getValidators return all validators with complete flag set to true', async () => {
+  const mockTransport = () =>
+    createTransport({
+      key: 'mock',
+      name: 'Mock Transport',
+      request: vi.fn(({ method, params }) => {
+        return mockData({
+          logs: {
+            '0x448fdaac1651fba39640e2103d83f78ff4695f95727f318f0f9e62c3e2d77a10': validInput,
+          },
+          method,
+          param: params,
+        });
+      }) as any,
+      type: 'mock',
+    });
+
+  const mockClient = createPublicClient({
+    transport: mockTransport,
+  });
+
+  const { validators, isComplete } = await getValidators(mockClient, {
+    rollupAddress: rollupAdminLogicAddress,
+  });
+
+  expect(validators).toEqual(['0x6a23CcC1c36D2aaA98AeF2a4471cf807DD22e45b']);
+  expect(isComplete).toBeTruthy();
+});
+
+it('getValidators return all validators with complete flag set to false', async () => {
+  const mockTransport = () =>
+    createTransport({
+      key: 'mock',
+      name: 'Mock Transport',
+      request: vi.fn(({ method, params }) => {
+        return mockData({
+          logs: {
+            '0x448fdaac1651fba39640e2103d83f78ff4695f95727f318f0f9e62c3e2d77a10': validInput,
+            '0x6e29af776e4b08f92b484a3d4ecc506a4b6455bbd335a2547c4e97d6151f588c': '0xdeadbeef',
+          },
+          method,
+          param: params,
+        });
+      }) as any,
```

Comment:
viem uses `as unknown as EIP1193RequestFn` so maybe we should do the same

https://github.com/wevm/viem/blob/main/src/clients/transports/createTransport.test.ts#L10C37-L10C68